### PR TITLE
feat(helm): set enableOCI in the UI when registering a repo

### LINKED ISSUE #12363: not able to add OCI docker hub repo into the argocd
is it possible to add OCI docker hub repo into the argocd?

when i tried to add the repo using secrets, its failed and not able to see any logs for that

here is my secret file

apiVersion: v1
kind: Secret
metadata:
  name: wrm6
  namespace: argocd
  labels:
    argocd.argoproj.io/secret-type: repository
stringData:
  enableOCI: "true"
  type: helm
  url: registry-1.docker.io/enghousenetworksbss/wrm

here is my repo failed 

image

### PR DESCRIPTION
**Pull request by alexef**
related: https://github.com/argoproj/argo-cd/issues/8548

fixes: https://github.com/argoproj/argo-cd/issues/12363

there seems to be a lot of confusion around how an OCI helm can be defined.

this changeset allows a flag to be set at repo creation on the UI (to be on par with manifest/api alternative)

Screenshot 2023-02-11 at 22 41 41
Screenshot 2023-02-11 at 22 41 17

Signed-off-by: Alex Eftimie <alex.eftimie@getyourguide.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: ui/src/app/settings/components/repos-list/repos-list.tsx [hunk 1]
```diff
@@ -173,7 +175,9 @@ export class ReposList extends React.Component<
             case ConnectionMethod.HTTPS:
                 const httpsValues = params as NewHTTPSRepoParams;
                 return {
-                    url: (!httpsValues.url && 'Repository URL is required') || (this.credsTemplate && !this.isHTTPSUrl(httpsValues.url) && 'Not a valid HTTPS URL'),
+                    url:
+                        (!httpsValues.url && 'Repository URL is required') ||
+                        (this.credsTemplate && !this.isHTTPSUrl(httpsValues.url) && !httpsValues.enableOCI && 'Not a valid HTTPS URL'),
```
Comment (Marvin9) [inline ui/src/app/settings/components/repos-list/repos-list.tsx:180]: Is it intended that URL shouldn't be validated if `enableOCI` is `true` and URL doesn't include `https` protocol?

Comment (alexef) [reply ui/src/app/settings/components/repos-list/repos-list.tsx:180]: yes. that's because internal logic expects the OCI url not to have a protocol

**Comment by alexef** [comment]
@crenshaw-dev I know you have 1M things on your plate, but please have a look at this small ui improvement

**Comment by crenshaw-dev** [comment]
@alexef could you take a look at lint again?

**Comment by alexef** [comment]
@crenshaw-dev lint looks better now, e2e tests don't, I doubt it's related

**Review by crenshaw-dev — APPROVED** [review]
Local test lgtm. Thanks, @alexef!

**Pull request merged**